src/extractor/extractor.cpp

```diff
@@ -409,33 +410,31 @@ std::shared_ptr<RestrictionMap> Extractor::LoadRestrictionMap()
   \brief Load node based graph from .osrm file
   */
 std::shared_ptr<util::NodeBasedDynamicGraph>
-Extractor::LoadNodeBasedGraph(std::unordered_set<NodeID> &barrier_nodes,
-                              std::unordered_set<NodeID> &traffic_lights,
+Extractor::LoadNodeBasedGraph(std::unordered_set<NodeID> &barriers,
+                              std::unordered_set<NodeID> &traffic_signals,
                               std::vector<QueryNode> &internal_to_external_node_map)
 {
-    std::vector<NodeBasedEdge> edge_list;
-
-    boost::filesystem::ifstream input_stream(config.output_file_name,
-                                             std::ios::in | std::ios::binary);
+    boost::filesystem::ifstream stream(config.output_file_name, std::ios::binary);
 
-    std::vector<NodeID> barrier_list;
-    std::vector<NodeID> traffic_light_list;
-    NodeID number_of_node_based_nodes = util::loadNodesFromFile(
-        input_stream, barrier_list, traffic_light_list, internal_to_external_node_map);
+    if (!stream)
+    {
+        throw util::exception("Unable to open " + config.output_file_name +
+                              " trying to read the node based graph");
+    }
 
-    util::SimpleLogger().Write() << " - " << barrier_list.size() << " bollard nodes, "
-                                 << traffic_light_list.size() << " traffic lights";
+    auto barriers_iter = inserter(barriers, end(barriers));
+    auto traffic_signals_iter = inserter(traffic_signals, end(traffic_signals));
 
-    // insert into unordered sets for fast lookup
-    barrier_nodes.insert(barrier_list.begin(), barrier_list.end());
-    traffic_lights.insert(traffic_light_list.begin(), traffic_light_list.end());
+    NodeID number_of_node_based_nodes = util::loadNodesFromFile(stream, //
+                                                                barriers_iter,
+                                                                traffic_signals_iter,
+                                                                internal_to_external_node_map);
```

Comment:
this is what I wanted to target: no need to create two vectors, transform to sets, clear and shrink vectors again